CI: pkg/alibabacloud/eni: ENISuite/TestPrepareIPAllocation is flaky

## CI failure

```
 --- FAIL: Test (0.04s)
    --- FAIL: Test/ENISuite (0.04s)
        --- FAIL: Test/ENISuite/TestPrepareIPAllocation (0.00s)
            node_test.go:159: 
                ... obtained int = 2
                ... expected int = 1
```

Seems similar to https://github.com/cilium/cilium/issues/21964 - cc @jaffcheng.

https://github.com/cilium/cilium/actions/runs/8016451004/job/21906345714

```
level=info msg="Synchronized ENI information" numInstances=3 numSecurityGroups=1 numVPCs=1 numVSwitches=2 subsys=eni
level=info msg="Discovered new CiliumNode custom resource" name=node3 subsys=ipam
level=warning msg="Unable to compute pending pods, will not surge-allocate" error="pod store uninitialized" instanceID=i-3 name=node3 subsys=ipam
level=info msg="Resolving IP deficit of node" available=0 availableForAllocation=0 emptyInterfaceSlots=1 instanceID=i-3 maxIPsToAllocate=1 name=node3 neededIPs=1 remainingInterfaces=1 selectedInterface= selectedPoolID= subsys=ipam used=0
level=info msg="No more IPs available, creating new ENI" instanceID=i-3 name=node3 securityGroupIDs="[sg-2]" subsys=ipam toAllocate=1 vSwitchID=vsw-2
level=info msg="Created new ENI" eniID=9c5030d9-6592-4a74-abea-247946f3912f instanceID=i-3 name=node3 securityGroupIDs="[sg-2]" subsys=ipam toAllocate=1 vSwitchID=vsw-2
level=info msg="Attached ENI to instance" eniID=9c5030d9-6592-4a74-abea-247946f3912f instanceID=i-3 name=node3 securityGroupIDs="[sg-2]" subsys=ipam toAllocate=1 vSwitchID=vsw-2
level=info msg="Synchronized ENI information for the corresponding instance" instance=i-3 numSecurityGroups=1 numVPCs=1 numVSwitches=2 subsys=eni
level=info msg="Synchronized ENI information" numInstances=3 numSecurityGroups=1 numVPCs=1 numVSwitches=2 subsys=eni
level=info msg="Discovered new CiliumNode custom resource" name=node1 subsys=ipam
level=info msg="Discovered new CiliumNode custom resource" name=node2 subsys=ipam
level=info msg="No more IPs available, creating new ENI" securityGroupIDs="[sg-1]" subsys=eni toAllocate=10 vSwitchID=vsw-1
level=warning msg="Unable to compute pending pods, will not surge-allocate" error="pod store uninitialized" instanceID=i-2 name=node2 subsys=ipam
level=info msg="Resolving IP deficit of node" available=0 availableForAllocation=0 emptyInterfaceSlots=2 instanceID=i-2 maxIPsToAllocate=8 name=node2 neededIPs=8 remainingInterfaces=2 selectedInterface= selectedPoolID= subsys=ipam used=0
level=info msg="No more IPs available, creating new ENI" instanceID=i-2 name=node2 securityGroupIDs="[sg-2]" subsys=ipam toAllocate=8 vSwitchID=vsw-2
level=info msg="Created new ENI" eniID=5444d730-8035-45da-b385-74a048ae8ded securityGroupIDs="[sg-1]" subsys=eni toAllocate=10 vSwitchID=vsw-1
level=info msg="Attached ENI to instance" eniID=5444d730-8035-45da-b385-74a048ae8ded securityGroupIDs="[sg-1]" subsys=eni toAllocate=10 vSwitchID=vsw-1
level=info msg="No more IPs available, creating new ENI" securityGroupIDs="[sg-1]" subsys=eni toAllocate=10 vSwitchID=vsw-1
level=info msg="Created new ENI" eniID=62bb3410-2f32-4f5a-ad82-9175e075825d securityGroupIDs="[sg-1]" subsys=eni toAllocate=10 vSwitchID=vsw-1
level=info msg="Attached ENI to instance" eniID=62bb3410-2f32-4f5a-ad82-9175e075825d securityGroupIDs="[sg-1]" subsys=eni toAllocate=10 vSwitchID=vsw-1
level=warning msg="Unable to compute pending pods, will not surge-allocate" error="pod store uninitialized" instanceID=i-1 name=node1 subsys=ipam
level=info msg="Resolving IP deficit of node" available=0 availableForAllocation=0 emptyInterfaceSlots=0 instanceID=i-1 maxIPsToAllocate=8 name=node1 neededIPs=8 remainingInterfaces=0 selectedInterface= selectedPoolID= subsys=ipam used=0
level=warning msg="Instance is out of interfaces" instanceID=i-1 name=node1 subsys=ipam
level=info msg="Created new ENI" eniID=34ee385d-5c15-4677-a891-9d084b0dca73 instanceID=i-2 name=node2 securityGroupIDs="[sg-2]" subsys=ipam toAllocate=8 vSwitchID=vsw-2
level=info msg="Attached ENI to instance" eniID=34ee385d-5c15-4677-a891-9d084b0dca73 instanceID=i-2 name=node2 securityGroupIDs="[sg-2]" subsys=ipam toAllocate=8 vSwitchID=vsw-2
level=info msg="Synchronized ENI information for the corresponding instance" instance=i-2 numSecurityGroups=1 numVPCs=1 numVSwitches=2 subsys=eni
level=info msg="Synchronized ENI information" numInstances=3 numSecurityGroups=1 numVPCs=1 numVSwitches=2 subsys=eni
level=info msg="Discovered new CiliumNode custom resource" name=node1 subsys=ipam
level=warning msg="Unable to compute pending pods, will not surge-allocate" error="pod store uninitialized" instanceID=i-1 name=node1 subsys=ipam
level=info msg="Resolving IP deficit of node" available=0 availableForAllocation=0 emptyInterfaceSlots=2 instanceID=i-1 maxIPsToAllocate=8 name=node1 neededIPs=8 remainingInterfaces=2 selectedInterface= selectedPoolID= subsys=ipam used=0
level=info msg="No more IPs available, creating new ENI" instanceID=i-1 name=node1 securityGroupIDs="[sg-1]" subsys=ipam toAllocate=8 vSwitchID=vsw-1
level=info msg="No more IPs available, creating new ENI" securityGroupIDs="[sg-1]" subsys=eni toAllocate=10 vSwitchID=vsw-1
level=info msg="Created new ENI" eniID=ab89b1cc-c5d5-4fbf-bd16-f64e8d0f2361 instanceID=i-1 name=node1 securityGroupIDs="[sg-1]" subsys=ipam toAllocate=8 vSwitchID=vsw-1
level=info msg="Attached ENI to instance" eniID=ab89b1cc-c5d5-4fbf-bd16-f64e8d0f2361 instanceID=i-1 name=node1 securityGroupIDs="[sg-1]" subsys=ipam toAllocate=8 vSwitchID=vsw-1
level=info msg="Created new ENI" eniID=3c58c84d-1628-44c1-b8cb-46fe4ae0e16a securityGroupIDs="[sg-1]" subsys=eni toAllocate=10 vSwitchID=vsw-1
level=info msg="Attached ENI to instance" eniID=3c58c84d-1628-44c1-b8cb-46fe4ae0e16a securityGroupIDs="[sg-1]" subsys=eni toAllocate=10 vSwitchID=vsw-1
level=info msg="Synchronized ENI information for the corresponding instance" instance=i-1 numSecurityGroups=1 numVPCs=1 numVSwitches=2 subsys=eni
--- FAIL: Test (0.04s)
    --- FAIL: Test/ENISuite (0.04s)
        --- FAIL: Test/ENISuite/TestPrepareIPAllocation (0.00s)
            node_test.go:159: 
                ... obtained int = 2
                ... expected int = 1
                
FAIL
coverage: 2.2% of statements
FAIL	github.com/cilium/cilium/pkg/alibabacloud/eni	0.221s
```

--- a/pkg/alibabacloud/eni/node_test.go
+++ b/pkg/alibabacloud/eni/node_test.go
@@ -143,11 +143,12 @@ func (e *ENISuite) TestPrepareIPAllocation(c *check.C) {
 	mngr, err := ipam.NewNodeManager(instances, k8sapi, metricsapi, 10, false, false)
 	c.Assert(err, check.IsNil)
 	c.Assert(mngr, check.Not(check.IsNil))
+	mngr.SetInstancesAPIReadiness(false) // to avoid the manager background jobs starting and racing us.
 
 	mngr.Upsert(newCiliumNode("node1", "i-1", "ecs.g7ne.large", "cn-hangzhou-i", "vpc-1"))
 	a, err := mngr.Get("node1").Ops().PrepareIPAllocation(log)
 	c.Assert(err, check.IsNil)
-	c.Assert(a.EmptyInterfaceSlots+a.IPv4.InterfaceCandidates, check.Equals, 2)
+	c.Assert(a.EmptyInterfaceSlots+a.IPv4.InterfaceCandidates, check.Equals, 2, check.Commentf("empty: %v, candidates: %v", a.EmptyInterfaceSlots, a.IPv4.InterfaceCandidates))
 
 	// create one eni
 	toAlloc, _, err := mngr.Get("node1").Ops().CreateInterface(context.Background(), &ipam.AllocationAction{
@@ -162,7 +163,7 @@ func (e *ENISuite) TestPrepareIPAllocation(c *check.C) {
 	// one eni left
 	a, err = mngr.Get("node1").Ops().PrepareIPAllocation(log)
 	c.Assert(err, check.IsNil)
-	c.Assert(a.EmptyInterfaceSlots+a.IPv4.InterfaceCandidates, check.Equals, 1)
+	c.Assert(a.EmptyInterfaceSlots+a.IPv4.InterfaceCandidates, check.Equals, 1, check.Commentf("empty: %v, candidates: %v", a.EmptyInterfaceSlots, a.IPv4.InterfaceCandidates))
 }
 
 func (e *ENISuite) TestNode_allocENIIndex(c *check.C) {

